reporter/src/main/kotlin/utils/FreemarkerTemplateProcessor.kt

```diff
@@ -54,13 +60,41 @@ class FreemarkerTemplateProcessor(
     companion object {
         const val OPTION_TEMPLATE_ID = "template.id"
         const val OPTION_TEMPLATE_PATH = "template.path"
+        const val OPTION_PROJECT_TYPES_AS_PACKAGES = "project.types.as.packages"
     }
 
     /**
      * Process all Freemarker templates referenced in "template.id" and "template.path" options and returns the
      * generated files.
      */
     fun processTemplates(input: ReporterInput, outputDir: File, options: Map<String, String>): List<File> {
+        val projectTypesAsPackages = options[OPTION_PROJECT_TYPES_AS_PACKAGES]?.split(",").orEmpty().toSet()
+        val projectsAsPackages = input.ortResult.getProjects().map { it.id }.filterTo(mutableSetOf()) {
+            it.type in projectTypesAsPackages
+        }
+
+        if (projectTypesAsPackages.isNotEmpty()) {
+            log.info { "Handling ${projectTypesAsPackages.size} of types $projectTypesAsPackages as packages." }
+        }
+
+        return processTemplatesInternal(
+            input = input.deduplicateProjectScanResults(projectsAsPackages),
```

Comment:
I think this feature is complex and could be difficult to debug, a test case would be much appreciated.

Reply:
I'd propose to add a test case which verifies that the `findings` related to the `target project types` do get `subtracted` from the scan result of `all other` projects. That is, the test case would verify the scan results contained in the `OrtResult`.

Would that test case address what you are missing @mnonnenmacher ?

Reply:
I think testing this on the level of `processTemplates()` would be more complete, but a test case for `deduplicateProjectScanResults()` would probably be more elegant. Both would be fine for me as most of the complexity is in the code of `deduplicateProjectScanResults()`.

Reply:
Done.